plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardRuntimeChangeListener.java

```diff
@@ -35,6 +36,51 @@
  */
 public class AppEngineStandardRuntimeChangeListener implements IFacetedProjectListener {
 
+  /** A Job to install out App Engine facet given that we've added a runtime. */
+  private static class InstallAppEngineFacetJob extends Job {
+    private final IFacetedProject facetedProject;
+    private final IRuntime newRuntime;
+
+    private InstallAppEngineFacetJob(IFacetedProject facetedProject,
+        IRuntime newRuntime) {
+      super(Messages.getString("appengine.add.facet.to.project",
+          facetedProject.getProject().getName())); // $NON-NLS$
+      this.facetedProject = facetedProject;
+      this.newRuntime = newRuntime;
+    }
+
+    /**
+     * Mark this job as a component update job. Useful for our tests to ensure project configuration
+     * is complete.
+     */
+    @Override
+    public boolean belongsTo(Object family) {
```

Comment:
Would it make sense to create an abstract parent job class that contains this method, and our jobs inherit from that, instead of defining it in each of them?